<?xml version="1.0"?>
<launch>
    <!--DBW Node -->
    <include file="$(find twist_controller)/launch/dbw.launch"/>

    <!--Waypoint Loader -->
    <include file="$(find waypoint_loader)/launch/waypoint_loader_site.launch"/>

    <!--Waypoint Follower Node -->
    <include file="$(find waypoint_follower)/launch/pure_pursuit.launch"/>

    <!--Waypoint Updater Node -->
    <include file="$(find waypoint_updater)/launch/waypoint_updater.launch"/>

    <!--Traffic Light Locations and Camera Config -->
    <param name="traffic_light_config" textfile="$(find tl_detector)/site_traffic_light_config.yaml" />

    <!--Traffic Light Detector Node -->
    <include file="$(find tl_detector)/launch/tl_detector_site.launch"/>

    <!--Camera Calibration YAML -->
    <param name="grasshopper_calibration_yaml" textfile="$(find styx)../../../data/grasshopper_calibration.yml" />

    <!--Camera Info Publisher -->
    <include file="$(find camera_info_publisher)/launch/camera_info_publisher.launch"/>

    <!--Save Camera Images for Training -->
    <param name="save_images" value="false" type="bool" />

    <!--Use Ground Truth -->
    <param name="use_ground_truth" value="false" type="bool" />

    <!--Distance ahead of vehicle to check for lights -->
    <param name="check_light_distance" value="100.0" type="double" />

    <!--Skip Images Count -->
    <param name="skip_images" value="0" type="int" />

    <!--lowest speed of the car in m/s-->
    <param name="min_speed" value="0.1" type="double" />

    <!--Throttle PID param p-->
    <param name="kp" value="0.3" type="double" />

    <!--Throttle PID param i-->
    <param name="ki" value="0.1" type="double" />

    <!--Throttle PID param d-->
    <param name="kd" value="0.0" type="double" />

    <!--Brake PID param p-->
    <param name="kp_brake" value="0.9" type="double" />

    <!--Brake PID param i-->
    <param name="ki_brake" value="0.001" type="double" />

    <!--Brake PID param d-->
    <param name="kd_brake" value="0.6" type="double" />

    <!--Min Throttle  -->
    <param name="min_throttle" value="0.0" type="double" />

    <!--Max Throttle from 0 - 1.0
    For Highway track 1.0 and for parking lot we might need to go to 0.2 -->
    <param name="max_throttle" value="1.0" type="double" />

    <!--Low Pass Filter tau-->
    <param name="lpf_tau" value="0.5" type="double" />

    <!--Low Pass Filter ts-->
    <param name="lpf_ts" value="0.02" type="double" />

    <!--Torque to apply brakes for a complete stop-->
    <param name="torque_complete_stop" value="400" type="double" />

    <!--Throttle amount, below which brakes can be applied -->
    <param name="throttle_limit_apply_brakes" value="0.1"
    type="double" />

    <!--Desired speed limit at which vehicle will do a complete stop-->
    <param name="complete_stop_speed" value="0.0"
    type="double" />
</launch>
